jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -159,47 +229,126 @@ def _on_spectral_unit_changed(self, *args):
                                self.spectral_unit.selected,
                                sender=self))
 
-    @observe('flux_unit_selected')
+    @observe('flux_unit_selected', 'sb_unit_selected')
     def _on_flux_unit_changed(self, *args):
+        flux_or_sb = None
+        current_y = self.spectrum_viewer.state.y_display_unit
 
-        yunit = _valid_glue_display_unit(self.flux_or_sb_unit.selected, self.spectrum_viewer, 'y')
+        data_collection_unit = ''
+        # need to determine the input spectrum units to disable the additional
+        # drop down and possiblity of translations in Specviz.
+        if self.app.data_collection[0] and self.app.config == 'specviz':
+            if check_if_unit_is_per_solid_angle(self.app.data_collection[0].get_object()._unit):
+                data_collection_unit = 'Surface Brightness'
+                self.specviz_disabler = 'Flux'
+            else:
+                self.specviz_disabler = 'Surface Brightness'
+                data_collection_unit = 'Flux'
```

Comment:
is this supposed to be Surface Brightness?

Reply:
The if statement checks if the data item's unit to see if it is per solid angle. If it is the we know to disable Flux. The else then would disable Surface Brightness. I did flip the order which with the variables are set from the if versus the else. I can flip this so it is more readable.

Reply:
ah, yes, that was what confused me (and the name `specviz_disabler` is still tripping me up as to what the value means)